app/models/course/assessment/submission.rb

```diff
@@ -91,6 +91,22 @@ def auto_grade!
     AutoGradingJob.perform_later(self)
   end
 
+  # Return the maximum step that the current submission can reach.
+  #
+  # @return [Integer] the maximum step.
+  def max_step
```

Comment:
@lowjoel Since we use numbered steps in controller too, I can't think anything better than numbers to indicate steps here. You have any suggestions ? 

The only issue here I think is that step is 1 based while index is 0 based, so `step - 1` represents current question and `step - 2` is the previous question (might be a bit confusing when first see it).